numpy/lib/function_base.py

```diff
@@ -4253,10 +4253,231 @@ def percentile(a, q, axis=None, out=None,
     >>> np.percentile(b, 50, axis=1, overwrite_input=True)
     array([ 7.,  2.])
     >>> assert not np.all(a == b)
+    """
+    return quantile(a, q=np.asarray(q)/100., axis=axis, weights=None,
+                    out=out, overwrite_input=overwrite_input,
+                    interpolation=interpolation, keepdims=keepdims)
+
 
+def quantile(a, q, axis=None, weights=None, out=None,
+             overwrite_input=False, interpolation='linear', keepdims=False):
+    """
+    Compute the qth quantile of the data along the specified axis,
+    where q = [0, 1].
+
+    Returns the qth quantiles(s) of the array elements.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array or object that can be converted to an array.
+    q : float in range of [0,100] (or sequence of floats)
+        Percentile to compute, which must be between 0 and 100 inclusive.
+    axis : {int, sequence of int, None}, optional
+        Axis or axes along which the percentiles are computed. The
+        default is to compute the percentile(s) along a flattened
+        version of the array. A sequence of axes is supported since
+        version 1.9.0.
+    weights : array_like, optional
+        An array of weights associated with the values in `a`. Each value in
+        `a` contributes to the average according to its associated weight.
+        The weights array can either be 1-D (in which case its length must be
+        the size of `a` along the given axis) or of the same shape as `a`.
```

Comment:
I think this should be relaxed to _"Either 1D, or such that `weights.ndim == a.ndim` and that `weights` and `a` are broadcastable"_